checks/network_checks.py

```diff
@@ -183,33 +187,57 @@ def _process_results(self):
 
                 threshold = instance.get('threshold', 1)
 
-                if len(self.statuses[instance_name][sc_name]) > window:
-                    self.statuses[instance_name][sc_name].pop(0)
+                if len(self.statuses[instance_key][sc_name]) > window:
+                    self.statuses[instance_key][sc_name].pop(0)
 
-                nb_failures = self.statuses[instance_name][sc_name].count(Status.DOWN)
+                nb_failures = self.statuses[instance_key][sc_name].count(Status.DOWN)
 
                 if nb_failures >= threshold:
-                    if self.notified.get((instance_name, sc_name), Status.UP) != Status.DOWN:
+                    if self.notified.get((instance_key, sc_name), Status.UP) != Status.DOWN:
                         event = self._create_status_event(sc_name, status, msg, instance)
-                        self.notified[(instance_name, sc_name)] = Status.DOWN
+                        self.notified[(instance_key, sc_name)] = Status.DOWN
                 else:
-                    if self.notified.get((instance_name, sc_name), Status.UP) != Status.UP:
+                    if self.notified.get((instance_key, sc_name), Status.UP) != Status.UP:
                         event = self._create_status_event(sc_name, status, msg, instance)
-                        self.notified[(instance_name, sc_name)] = Status.UP
+                        self.notified[(instance_key, sc_name)] = Status.UP
 
                 if event is not None:
                     self.events.append(event)
 
-            # The job is finished here, this instance can be re processed
-            if instance_name in self.jobs_status:
-                del self.jobs_status[instance_name]
-
-            # if an exception happened, log it
-            if instance_name in self.jobs_results:
-                ret = self.jobs_results[instance_name].get()
-                if isinstance(ret, Exception):
-                    self.log.exception("Exception in worker thread: {0}".format(ret))
-                del self.jobs_results[instance_name]
+            self._clean_job(instance_key)
+
+    def _get_instance_key(self, instance):
```

Comment:
Shouldn't the instance key generation be in `snmp.py` instead? Because we enforce the existence and uniqueness of the `name` attribute in the `__init__` method of `NetworkCheck` (https://github.com/DataDog/dd-agent/blob/jaime/snmpv3fix/checks/network_checks.py#L72-L78), so we'll always return the `name` of the instance here.

Whereas what would probably make more sense would be to generate a unique `name` there: https://github.com/DataDog/dd-agent/blob/jaime/snmpv3fix/checks.d/snmp.py#L55-L57
and then use the `name` as an identifier of the instance in the check.

The method of generation of the unique name can also depend on the specifics of each kind of network check IMO.